patches/chrome-android-java-res-layout-new_tab_page_incognito.xml.patch

```diff
@@ -0,0 +1,25 @@
+diff --git a/chrome/android/java/res/layout/new_tab_page_incognito.xml b/chrome/android/java/res/layout/new_tab_page_incognito.xml
+index b294fa17a33e3c089babc00ca4d17b2a62a9e8f3..2003f8ad6712db98e235d75ed763f3e703863540 100644
+--- a/chrome/android/java/res/layout/new_tab_page_incognito.xml
++++ b/chrome/android/java/res/layout/new_tab_page_incognito.xml
+@@ -3,7 +3,7 @@
+      Use of this source code is governed by a BSD-style license that can be
+      found in the LICENSE file. -->
+ 
+-<org.chromium.chrome.browser.ntp.IncognitoNewTabPageView
++<org.chromium.chrome.browser.ntp.BraveIncognitoNewTabPageView
+     xmlns:android="http://schemas.android.com/apk/res/android"
+     xmlns:tools="http://schemas.android.com/tools"
+     android:id="@+id/ntp_content"
+@@ -82,8 +82,10 @@
+                 android:textAppearance="@style/TextAppearance.IncognitoNewTabLearnMoreLinkModern"
+                 android:lineSpacingExtra="@dimen/md_incognito_ntp_line_spacing" />
+ 
++	    <include layout="@layout/brave_ddg_offer_link" android:layout_height="wrap_content" android:layout_width="match_parent" />
```

Comment:
Curious we can do this by programatically instead of using layout xml. If so we can minimize patch?

Reply:
Can we implement `BraveIncognitoNewTabPageView` as a super class of `IncognitoNewTabPageView`?
If we can, I think it can also reduce more patch files.

Reply:
@simonhong It doesn't seem like a super class would work here, since I need to override a function. But maybe I'm misunderstanding?

Reply:
`IncognitoNewTabPageView` `onFinishInflate` calls `super.onFinishInflate` so I think a superclass would work and would only be a single patch to IncognitoNewTabPageView to change the superclass

Reply:
actually I think maybe we can simplify this by not overriding `IncognitoNewTabPageView` at all and just adding a view class here in the xml file for ddg

Reply:
Yup, good idea - fixed.